Adding a "Debugging" section to FAQ describing conda envs (#309)
showyourwork/showyourwork: docs/faqs.rst

in which case the ``git clone`` silently fails and no ``master`` branch is created.
If you run into this error, delete or comment out the ``overleaf:`` section of the ``environment.yml``
workflow config and re-run the workflow, or simply upgrade |showyourwork|.

> Debugging in |showyourwork|'s conda environment
> --------------------

Reviewer: ```suggestion
Debugging in |showyourwork|'s conda environment
-----------------------------------------------
```

> Sometimes when debugging it can be helpful to bipass the ``showyourwork build`` command and

Reviewer: ```suggestion
Sometimes when debugging it can be helpful to bypass the ``showyourwork build`` command and
```

run your code manually in the same environment |showyourwork| is using. To do that, you
need to access the `conda environment <https://docs.conda.io/en/latest/>`_ that Snakemake
creates.

If conda is not your default package manager you may need to activate it using
``conda activate``. Once activated, you can view the environments that exist on your

> computer using ``conda env list``. The output will look something like this:

Reviewer: ```suggestion
computer using ``conda env list``. On a Unix machine, the output will look something like this:
```

.. code-block:: text

>   base     /Users/rwegener/miniconda3
>            /Users/rwegener/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_

Reviewer: ```suggestion
  base     /path/to/miniconda3
           /path/to/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_
           
where "`/path/to/`" is often your home directory or wherever you installed Anaconda and cloned your article's repository.
```

The environment for the |showyourwork| project is not given a name, but the path
to the environment should be listed. The environment path will be something of the form:
`the path to your article``` + ``.snakemake/conda`` + `a long
hexadecimal string (called a hash)`. In this example the environment path is

> ``/Users/rwegener/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_``.

Reviewer: ```suggestion
``/path/to/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_``.
```

> To activate the environment, copy and paste that full path and run ``conda activate MY_ENV_PATH``.

Reviewer: ```suggestion
To activate the environment, copy and paste that full path and run ``conda activate <full path>``.
```

For example:

.. code-block:: text

>   conda activate /Users/rwegener/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_

Reviewer: ```suggestion
  conda activate /path/to/my-article/.snakemake/conda/28e8667c7a3a18371030aff088ceb5c5_
```

> From this point anything you run using the ``python`` command should use the same conda

Reviewer: ```suggestion
From this point, anything you run using the ``python`` command should use the same conda
```
